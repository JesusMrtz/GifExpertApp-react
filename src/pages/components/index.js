export * from './AddCategoryComponent';
export * from './GifGridComponent';
export * from './GifItemComponent';