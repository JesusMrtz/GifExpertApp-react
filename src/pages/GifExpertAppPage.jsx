import { useState } from "react";
import { AddCategoryComponent, GifGridComponent } from "./components";

export function GifExpertAppPage() {
    const [ categories, setCategories ] = useState([]);

    function onAddCategory( category ) {
        const existCategory = categories.some( (c) => c.toLowerCase() === category.toLowerCase() );

        if ( existCategory ) return;
        setCategories([category, ...categories])
    }

    return (
        <>
            <h1>GifExpertApp</h1>
            <AddCategoryComponent  onNewCategory={ onAddCategory } />
            {
                !categories.length && (<strong className="message">Vamos a buscar imagenes üçî</strong>)
            }
                {/* Vamos a barrer cada unos de los elementos del arreglo */}
                {/* Necesitamos un key cuando construyamos de manera d√≠namica */}
                {
                    categories.map(( category, index ) => (
                        // No es recomendable utilizar el "index" como key porque react puede confundirse
                        // Este c√≥digo comentado con el key de index.. me da error
                        // <GifGridComponent key={ category } category={ category } />
                        
                        <GifGridComponent key={ category } category={ category } />
                    ))
                }
        </>
    );
}