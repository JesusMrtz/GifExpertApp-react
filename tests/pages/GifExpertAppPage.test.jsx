import { fireEvent, render, screen } from "@testing-library/react";
import { GifExpertAppPage } from '../../src/pages/GifExpertAppPage';


describe(`Pruebas en la page <GifExpertAppPage />`, () => {
    const newCategory = 'Saitama';

    it('Creamos el snapshot', () => {
        const { container } = render(<GifExpertAppPage />);
        expect( container ).toMatchSnapshot();
    });


    it('Verificar que no exista valores repetidos en el arreglo de búsquedas', async () => {
        const { container } = render(<GifExpertAppPage />);
        const input = screen.getByRole('textbox'); 
        const form = screen.getByRole('form'); 

        //disparo los eventos para agregar una categoría
        await fireEvent.input(input, { target: { value: newCategory } });
        await fireEvent.submit(form);//agrega Saitama

           //Intento agregar la misma categoria
        await fireEvent.input(input, { target: { value: newCategory } });
        await fireEvent.submit(form); //intento agregar Saitama nuevamente

        // screen.debug();

           //espero que no agregue la misma categoría y sólo esté agregada la primera vez
           expect(screen.getAllByRole('heading', { level: 1 }).length).toBe(1);
    });
});