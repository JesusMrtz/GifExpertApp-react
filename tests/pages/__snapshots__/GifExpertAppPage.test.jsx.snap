// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en la page <GifExpertAppPage /> Creamos el snapshot 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    role="form"
  >
    <input
      placeholder="Buscar GIF's"
      type="text"
      value=""
    />
  </form>
  <strong
    class="message"
  >
    Vamos a buscar imagenes üçî
  </strong>
</div>
`;
